### PR TITLE
8233790: Forward output from heap dumper to jcmd/jmap

### PR DESCRIPTION
**Pull request by buddyliao**
Backport of the heap dump enhancement: Forward output from heap dumper to jcmd/jmap

Patch did not apply cleanly, I had to resolve/modify
src/hotspot/share/services/diagnosticCommand.cpp
src/hotspot/share/services/heapDumper.cpp
src/hotspot/share/services/heapDumper.hpp

since patch 7cbb67a3f8adc83a5b51c092a66480d7b22a6bea is backport before this one, so this patch can't backport cleanly

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8233790](https://bugs.openjdk.java.net/browse/JDK-8233790): Forward output from heap dumper to jcmd/jmap

### Reviewers
 * [Paul Hohensee](https://openjdk.java.net/census#phh) (@phohensee - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jdk11u-dev pull/256/head:pull/256` \
`$ git checkout pull/256`

Update a local copy of the PR: \
`$ git checkout pull/256` \
`$ git pull https://git.openjdk.java.net/jdk11u-dev pull/256/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 256`

View PR using the GUI difftool: \
`$ git pr show -t 256`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jdk11u-dev/pull/256.diff">https://git.openjdk.java.net/jdk11u-dev/pull/256.diff</a>

</details>

### CODE REVIEW

**Review by phohensee — APPROVED** [review]

**Comment by linzang** [comment]
I have helped to add "jdk11u-fix-request" label at https://bugs.openjdk.java.net/browse/JDK-8233790. 
Thanks!

**Comment by phohensee** [comment]
I'll push as soon as I get Maintainer approval.

**Comment by buddyliao** [comment]
Would anyone else could help me to do this review? Thank you

**Comment by RealCLanger** [comment]
As Paul didn't react so far, I can sponsor this.
/sponsor

**Pull request closed without merging**
